bot/exts/code_jams/_flows.py

```diff
@@ -85,6 +85,79 @@ async def deletion_flow(
         await role.delete(reason="Code Jam ended.")
 
 
+async def add_flow(
+        bot: SirRobin,
+        team_name: str,
+        ctx: commands.Context,
+        member: discord.Member,
+        is_leader: bool = False
+) -> None:
+    """Add a member to the Code Jam and assign the roles accordingly."""
+    # Check if the user is not already a participant
+    try:
+        team = await bot.code_jam_mgmt_api.get(
+            f"users/{member.id}/current_team",
+            raise_for_status=True
+        )
+    except ResponseCodeError as err:
+        if err.response.status == 404:
+            # The user is not a participant, so the flow will proceed,
+            # and add the user to the database if it does not exist yet.
+            try:
+                await bot.code_jam_mgmt_api.post(f"users/{member.id}", raise_for_status=True)
+            except ResponseCodeError as err:
+                if err.response.status != 400:
+                    log.error(f"Something went wrong: {err}")
+            try:
+                team_to_move_in = await bot.code_jam_mgmt_api.get(
+                    "teams/find",
+                    params={"name": team_name},
+                    raise_for_status=True
+                )
+            except ResponseCodeError as err:
+                if err.response.status == 404:
+                    await ctx.send(f":x: Team `{team_name}` does not exist in the current jam!")
+                else:
+                    await ctx.send("Something went wrong while processing the request! We have notified the team!")
```

Comment:
Note that it *is the team* that is using this command. I suppose this still satisfies the purpose of getting us to check the logging 😄.

Reply:
Actually, here, by team, I meant the Python Discord team.